File: 7. Classes Abstratas-Sobrescrita e Polimorfismo/Funcionarios(C++)/Funcionario.cpp
#include "Funcionario.hpp"
Funcionario::Funcionario()
{
    this->nome = "";
}
Funcionario::Funcionario(std::string nome)
{
    this->nome = nome;
}
std::string Funcionario::get_nome()
{
    return this->nome;
}
Funcionario::~Funcionario()
{
}

File: Hospital-Parte-2/Funcionario.java
public abstract class Funcionario {
    protected String _nome;
    protected String _cpf;
    protected double _total_recebido;

    public Funcionario(String nome, String cpf) {
        this._total_recebido = 0;
        this._nome = nome;
        this._cpf = cpf;
    }

    public abstract void adicionaPagamento();

    public String getNome() {
        return _nome;
    }

    public String getCpf() {
        return _cpf;
    }

    public double getTotalRecebido() {
        return _total_recebido;
    }
}
